### ISSUE TITLE
[ruff-format] flake8-class-newline

### ISSUE THREAD

**Issue by oysstu** [astral-sh/ruff#8604]
Working on a codebase where [flake8-class-newline](https://pypi.org/project/flake8-class-newline/) is enforced. To quote that module:

> [PEP8](https://www.python.org/dev/peps/pep-0008/#blank-lines) says we should surround every class method with a single blank line. However flake8 is ambiguous about the first method having a blank line above it.

The ruff formatter currently removes leading spaces after a class definition if there is no docstring. Honestly, I've seen more Python classes without a leading space before the first method than ones with a leading space, so I don't think adding a space should necessary be the default, but it would be nice to have an option for having it (or ignoring it if it exists).

Class without newline (current ruff formatting)
```python
class AClassWithoutANewLine(object):
    def a_method(self):
        return 'a_value'
```

Class with newline (flake8-class-newline expected formatting)
```python
class AClassWithoutANewLine(object):

    def a_method(self):
        return 'a_value'
```

When a docstring is present, ruff currently adds a space between the docstring and first method, which passes the flake8-class-newline linter.

**Comment by zanieb** [comment]
Thanks for the report; this is a duplicate of https://github.com/astral-sh/ruff/issues/8566 and I think the reasoning there stands. Happy to hear additional thoughts though.

**Comment by oysstu** [comment]
Aha, didn't find that issue when searching. Fair enough, I wish the PEP8 would be more explicit regarding this
